source/fundamentals/crud/compound-operations.txt

```diff
@@ -0,0 +1,313 @@
+===================
+Compound Operations
+===================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide you will learn how to perform **compound operations** with the
+MongoDB Java driver. 
+
+Compound operations consist of a read and write operation performed as one
+**atomic operation**. An atomic operation is an operation which either completes
+entirely, or does not complete at all. Atomic operations cannot partially complete.
+
+Atomic operations can help you avoid **race conditions** in your code. A
+race condition occurs when your code's behavior is dependent on the order of
+uncontrollable events.
+
+MongoDB supports the following compound operations:
+
+- Find and update one document
+- Find and replace one document
+- Find and delete one document
+
+If you need to perform more complex tasks atomically, such as reading and
+writing to more than one document, use **transactions**. Transactions are a
+feature of MongoDB and other databases that lets you define an arbitrary
+sequence of database commands as an atomic operation.  
+
+For more information on atomic operations and atomicity, see 
+:manual:`the MongoDB manual entry for atomicity and transactions </core/write-operations-atomicity/>`.
+
+For more information on transactions, 
+:manual:`see the MongoDB manual entry for transactions </core/transactions/>`.
+
+How to Use Compound Operations
+------------------------------
+
+This section shows how to use each compound operation with the MongoDB Java Driver.
```

Comment:
```suggestion

```

Suggestion: Replace this with the part in your intro about the compound operations in the java driver

Reply:
Response to this edit is in the thread for https://github.com/mongodb/docs-java/pull/80#discussion_r634413543